src/newgrf_industries.cpp

```diff
@@ -575,13 +606,28 @@ void IndustryProductionCallback(Industry *ind, int reason)
 		if (tgroup == NULL || tgroup->type != SGT_INDUSTRY_PRODUCTION) break;
 		const IndustryProductionSpriteGroup *group = (const IndustryProductionSpriteGroup *)tgroup;
 
-		bool deref = (group->version == 1);
+		bool deref = (group->version >= 1);
 
-		for (uint i = 0; i < 3; i++) {
-			ind->incoming_cargo_waiting[i] = Clamp(ind->incoming_cargo_waiting[i] - DerefIndProd(group->subtract_input[i], deref) * multiplier, 0, 0xFFFF);
-		}
-		for (uint i = 0; i < 2; i++) {
-			ind->produced_cargo_waiting[i] = Clamp(ind->produced_cargo_waiting[i] + max(DerefIndProd(group->add_output[i], deref), 0) * multiplier, 0, 0xFFFF);
+		if (group->version < 2) {
```

Comment:
Thinking about this again, maybe there should still be a choice between direct/indirect cargo types specification here. I think @andythenorth talked about the indirection being a bother and not useful?

One idea would be to let the lowest bit specify whether indirection is used, and the next whether long cargo lists are used, such that:
- v0 = direct cargo spec, short lists
- v1 = indirect cargo spec, short lists
- v2 = direct cargo spec, long lists
- v3 = indirect cargo spec, long lists.

Reply:
Talked to Andy about this, no change to the spec/code is needed here.